src/slam_solver.h

```diff
@@ -0,0 +1,22 @@
+/* Extends ScanSolver to allow different solvers to publish graph visualizations differently */
+#ifndef SLAM_SOLVER_H
+#define SLAM_SOLVER_H
+
+#include <open_karto/Mapper.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <string>
+
+namespace karto {
+
+class SLAMSolver : public karto::ScanSolver
```

Comment:
I thought for a while if there wasn't a better name than `SLAMSolver`, since what it adds to `ScanSolver` is exclusively graph visualization, but I could not find anything. Let me know if you have ideas